### ISSUE TITLE
[BUG] SyncCopyFromUriAsync() requires a SAS uri while StartCopyFromUriAsync() does not

### ISSUE THREAD

**Issue by ig-sinicyn** [Azure/azure-sdk-for-net#32028]
### Library name and version

Azure.Storage.Blobs 12.14.1

### Describe the bug

We are trying to use SyncCopyFromUriAsync() as a replacement of StartCopyFromUriAsync() / WaitForCompletionAsync() pair.
(as a note: we do know about 256mb limit and our files are known to be less then 1 mb).

We have a working code like
```cs
	// Copy op: ok
	var sourceClient = containerClient.GetBlobClient("source.txt");
	var destinationClient = containerClient.GetBlobClient("destination.txt");
	//... 
	var copyOp = await destinationClient.StartCopyFromUriAsync(sourceClient.Uri);
	await copyOp.WaitForCompletionAsync();
```
and it breaks if we do use SyncCopyFromUriAsync() instead:
```cs
	// Sync copy: throws CannotVerifyCopySource error
	var sourceClient = containerClient.GetBlobClient("source.txt");
	var destinationClient = containerClient.GetBlobClient("destination.txt");
	//... 
	await destinationClient.SyncCopyFromUriAsync(sourceClient.Uri);

	// Sync copy with SAS: ok
	//await destinationClient.SyncCopyFromUriAsync(sourceClient.GenerateSasUri(BlobSasPermissions.Read, DateTimeOffset.UtcNow.AddMinutes(5)));
```

### Expected behavior

We do expect the SyncCopyFromUriAsync() should mirror the StartCopyFromUriAsync() contract and should accept non-SAS uris for blobs from the same container. 

### Actual behavior

 The SyncCopyFromUriAsync() throws RequestFailedException:
 ```
 Azure.RequestFailedException : Server failed to authenticate the request. Please refer to the information in the www-authenticate header.
RequestId:5b205f18-901e-0049-5837-e9557d000000
Time:2022-10-26T12:36:25.0532752Z
Status: 401 (Server failed to authenticate the request. Please refer to the information in the www-authenticate header.)
ErrorCode: CannotVerifyCopySource
 ```
 while trying to copy blob from the same container

### Reproduction Steps

```cs
	public sealed class BlobStorageTroubleshootingTests1
	{
		private const string StorageConnectionString = "DefaultEndpointsProtocol=https;AccountName=...;AccountKey=...;EndpointSuffix=core.windows.net";

		[Test]
		public async Task TestSyncCopy_Ok()
		{
			var blobServiceClient = new BlobServiceClient(StorageConnectionString);
			var containerClient = blobServiceClient.GetBlobContainerClient("delete-me-" + Guid.NewGuid().ToString("N"));
			var sourceClient = containerClient.GetBlobClient("source.txt");
			var destinationClient = containerClient.GetBlobClient("destination.txt");

			var exists = await containerClient.ExistsAsync();
			if (!exists)
				await containerClient.CreateAsync();

			try
			{
				using var content = new MemoryStream(Encoding.UTF8.GetBytes("Test content"));

				await sourceClient.UploadAsync(content);

				// Sync copy: throws CannotVerifyCopySource error
				await destinationClient.SyncCopyFromUriAsync(sourceClient.Uri);

				// Sync copy with SAS: ok
				//await destinationClient.SyncCopyFromUriAsync(sourceClient.GenerateSasUri(BlobSasPermissions.Read, DateTimeOffset.UtcNow.AddMinutes(5)));

				using var download = (await destinationClient.DownloadStreamingAsync()).Value;
				var expectedText = await new StreamReader(download.Content).ReadToEndAsync();

				Assert.AreEqual(expectedText, "Test content");
			}
			finally
			{
				if (!exists)
					await containerClient.DeleteAsync();
			}
		}

		[Test]
		public async Task TestCopyOp_Ok()
		{
			var blobServiceClient = new BlobServiceClient(StorageConnectionString);
			var containerClient = blobServiceClient.GetBlobContainerClient("delete-me-" + Guid.NewGuid().ToString("N"));
			var sourceClient = containerClient.GetBlobClient("source.txt");
			var destinationClient = containerClient.GetBlobClient("destination.txt");

			var exists = await containerClient.ExistsAsync();
			if (!exists)
				await containerClient.CreateAsync();

			try
			{
				using var content = new MemoryStream(Encoding.UTF8.GetBytes("Test content"));

				await sourceClient.UploadAsync(content);

				// Copy op: ok
				var copyOp = await destinationClient.StartCopyFromUriAsync(sourceClient.Uri);
				await copyOp.WaitForCompletionAsync();

				using var download = (await destinationClient.DownloadStreamingAsync()).Value;
				var expectedText = await new StreamReader(download.Content).ReadToEndAsync();

				Assert.AreEqual(expectedText, "Test content");
			}
			finally
			{
				if (!exists)
					await containerClient.DeleteAsync();
			}
		}
	}
```

### Environment

```
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.402\

Host:
  Version:      6.0.10
  Architecture: x64
  Commit:       5a400c212a

.NET SDKs installed:
  3.1.424 [C:\Program Files\dotnet\sdk]
  6.0.203 [C:\Program Files\dotnet\sdk]
  6.0.305 [C:\Program Files\dotnet\sdk]
  6.0.402 [C:\Program Files\dotnet\sdk]
```

**Comment by navba-MSFT** [comment]
@ig-sinicyn Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@ig-sinicyn I looked at the backend logs for the below error that you had encountered with SyncCopyFromUriAsync():

```
Azure.RequestFailedException : Server failed to authenticate the request. Please refer to the information in the www-authenticate header.
RequestId:5b205f18-901e-0049-5837-e9557d000000
Time:2022-10-26T12:36:25.0532752Z
Status: 401 (Server failed to authenticate the request. Please refer to the information in the www-authenticate header.)
ErrorCode: CannotVerifyCopySource
```

The SyncCopyFromUriAsync function uses [**Copy Blob Rest API**](https://learn.microsoft.com/en-us/rest/api/storageservices/copy-blob-from-url) underneath. So this behavior seems to be by-design and the same has been documented. 

>The source for a Copy Blob From URL operation can be any committed block blob in any Azure storage account **that's either public or authorized with a shared access signature.**

More Info:
x-ms-copy-source:name | Required. Specifies the URL of the source blob. The value can be a URL of up to 2 kibibytes (KiB) in length that specifies a blob. _The value should be URL-encoded as it would appear in a request URI. The source blob must either be public or be authorized via a shared access signature. If the source blob is public, no authorization is required to perform the operation._
-- | --

So if you don't wish to provide the SAS token, try to make the Access Level on the container to **public**. 
I tried to make the access level of container to public (from private) it worked without the SAS token.
